doc/manual/src/contributing/hacking.md

```diff
@@ -46,6 +46,12 @@ To build Nix itself in this shell:
 [nix-shell]$ make -j $NIX_BUILD_CORES
 ```
 
+> **Note**
+>
+> If using zsh and `./configure` is failing, use `${=configureFlags}` in place
+> of `$configureFlags` so the configure flags are expanded to multiple words as
+> they are in `sh` compatible shells.
```

Comment:
https://github.com/NixOS/nix/pull/8455#issuecomment-1579660935 Then the wording is misleading here. It reads to me as "using this syntax is the `sh` compatible way of expanding space-delimited strings into lists".

The entire premise may be off here. `nix develop` does some magic to hook you into the builder executable that happens to be `bash`. Assuming the user essentially does
```console
nix develop
[nix-shell]$ zsh
```
is too much for this manual to take care of. It would be in the user's responsibility to know how to pass around environment  variables in their shell of choice. Please point out if I'm completely mistaken here. Otherwise I suggest closing this PR.

Reply:
Use of direnv does not seem that unreasonable of a use case and I learned about this particularity of word expansion the day I filed this PR after using zsh as my main shell for five years. It's not a terribly obvious thing. But I understand your view as well.